crates/revm/src/context/evm_context.rs

```diff
@@ -0,0 +1,406 @@
+use super::inner_evm_context::InnerEvmContext;
+use crate::{
+    db::Database,
+    interpreter::{
+        return_ok, CallInputs, Contract, Gas, InstructionResult, Interpreter, InterpreterResult,
+    },
+    primitives::{Address, Bytes, EVMError, Env, HashSet, U256},
+    ContextPrecompiles, FrameOrResult, CALL_STACK_LIMIT,
+};
+use core::{
+    fmt,
+    ops::{Deref, DerefMut},
+};
+use std::boxed::Box;
+
+//#[derive(Debug)]
+pub struct EvmContext<DB: Database> {
+    pub inner: InnerEvmContext<DB>,
```

Comment:
is the wrapper type necessary?
can we move precompiles into evmcontextInner or does this have an advantage?